packages/web-twig/src/Resources/components/FieldGroup/README.md

```diff
@@ -0,0 +1,127 @@
+to the developer to configure the child fields correctly.**
+
+👉 The FieldGroup component implements the `<fieldset>` HTML element. Read more about the advantages and limitations in
+the [`web` implementation][gh-web-field-group-html] of `FieldGroup`.
+
+## Required Fields
+
+To render FieldGroup as required, add the `isRequired` prop:
+
+```twig
+<FieldGroup id="example-field-group" label="Label" isRequired>
+  <!-- Form fields… -->
+</FieldGroup>
+```
+
+⚠️ The `isRequired` prop is only used to indicate visually that all fields in the group are required. The individual
+fields themselves need to be marked as required using the `isRequired` prop.
+
+## Hidden Label
+
+To visually hide the FieldGroup label, add the `isLabelHidden` prop:
+
+```twig
+<FieldGroup id="example-field-group" label="Label" isLabelHidden>
+    <!-- Form fields… -->
+</FieldGroup>
+```
+
+⚠️ Remember the `label` prop should be always set to provide an accessible label for the group.
+
+## Helper Text
+
+To render helper text, add the `helperText` prop:
+
+```twig
+<FieldGroup helperText="Helper text" id="example-field-group" label="Label">
+    <!-- Form fields… -->
+</FieldGroup>
+```
+
+## Fluid Width
+
+To make the FieldGroup component fluid, add the `isFluid` prop:
+
+```twig
+<FieldGroup id="example-field-group" label="Label" isFluid>
+    <!-- Form fields… -->
+</FieldGroup>
+```
+
+## Disabled State
+
+To render FieldGroup as required, add the `isDisabled` prop:
+
+```twig
+<FieldGroup id="example-field-group" label="Label" isDisabled>
+    <!-- Form fields… -->
+</FieldGroup>
+```
+
+⚠️ Remember to also disable all fields in the group using the `isDisabled` prop.
+
+👉 Read more about the disabled state in the [`web` implementation][gh-web-field-group-disabled] of `FieldGroup`.
+
+## Validation States
+
+Just like any other form component in Spirit, FieldGroup implements the
+[Validation state dictionary][dictionary-validation].
+
+Validation states can be presented either by adding the `validationState` attribute, or by adding a JS interaction class
+on the native HTML input when controlled by JavaScript (`has-success`, `has-warning`, `has-danger`).
+
+When validated on server:
+
+```twig
+<FieldGroup
+  id="example-field-group"
+  label="Label"
+  validationState="success"
+  validationText="Success validation message"
+/>
+```
+
+## API
+
+| Prop name               | Type                                           | Default | Required | Description                                |
+| ----------------------- | ---------------------------------------------- | ------- | -------- | ------------------------------------------ |
+| `helperText`            | `string`                                       | `null`  | no\*\*   | Custom helper text                         |
+| `form`                  | `string`                                       | `null`  | no       | Parent form ID                             |
```

Comment:
🤔 `form` and `name` props are listed in the API docs here (so developers know they can use them), but they are just passed through the `_allowedAttributes` of `mainProps()` — is it OK? Or should I move them into the `props` object?

Reply:
Well, it was not a yes-or-no question 😄 but I checked other Twig components and most of them (though not all 🙈) list the allowed HTML attributes in the API table. Twig is different from React in this respect so I'd keep the allowed attributes listed everywhere.